ui/components/MeshModelComponent.js

```diff
@@ -1,23 +1,211 @@
 import {  withStyles } from '@material-ui/core'
 import { withSnackbar } from 'notistack';
-import React from 'react'
+import React, { useState, useEffect } from 'react'
+import MUIDataTable from 'mui-datatables';
+import { TableCell, Tooltip, TableSortLabel } from '@material-ui/core';
+import { getComponentsDetail, getModelsDetail } from '../api/meshmodel'
+
 const meshmodelStyles = (theme) => ({
-  wrapperClass : {
-    padding : theme.spacing(2),backgroundColor : theme.palette.secondary.elevatedComponents,height : "60vh",marginTop : theme.spacing(2),borderRadius : theme.spacing(1),
+  wrapperClss : {
+    flexGrow : 1,
+    maxWidth : '100%',
+    height : 'auto',
+  },
+  tab : {
+    minWidth : 40,
+    paddingLeft : 0,
+    paddingRight : 0,
+    "&.Mui-selected" : {
+      color : theme.palette.type === 'dark' ? "#00B39F" : theme.palette.primary,
+    },
+  },
+  tabs : {
+    "& .MuiTabs-indicator" : {
+      backgroundColor : theme.palette.type === 'dark' ? "#00B39F" : theme.palette.primary,
+    },
   },
   dashboardSection : {
-    padding : theme.spacing(2),borderRadius : 4,height : "100%",overflowY : "scroll"
+    padding : theme.spacing(2), borderRadius : 4,height : "100%",overflowY : "scroll"
   },
 
+
 })
-const MeshModelComponent = ({ showMeshModelSummary,classes }) => {
+
+const MeshModelComponent = ({ type, classes }) => {
+  const [resourcesDetail, setResourcesDetail] = useState();
+  const [isRequestCancelled, setRequestCancelled] = useState(false);
+  const [page, setPage] = useState(1);
+
+  const getModels = async (page) => {
+    try {
+      const data = await getModelsDetail(page);
+      if (!isRequestCancelled) {
+        setResourcesDetail(data);
+      }
+    } catch (error) {
+      console.error('Failed to fetch models:', error);
+    }
+  };
+
+  const getComponents = async (page) => {
+    try {
+      const data = await getComponentsDetail(page);
+      if (!isRequestCancelled) {
+        setResourcesDetail(data);
+      }
+    } catch (error) {
+      console.error('Failed to fetch components:', error);
+    }
+  };
+
+  useEffect(() => {
+
+    setRequestCancelled(false);
+
+
+    if (type === 'models') {
+      getModels(page);
+    } else if (type === 'components') {
+      getComponents(page);
+    }
+
+    return () => {
+      setRequestCancelled(true);
+    };
+  }, [type, page]);
+
+
+
+
+  const smi_columns = [
```

Comment:
should it be renamed to a meshmodel construct?

Reply:
is there any scope of re-usability of these constructs across our datatables?

Reply:
Yep, it should be renamed. 

Regarding re usability, I will make a separate PR since we are using data tables in multiple components.  Each component utilizes different configuration so i will figure out a way of standardizing the configurations.